### ISSUE TITLE
✨ Test re-enabled dark mode

### ISSUE THREAD

**Issue by substrae** [GCTC-NTGC/gc-digital-talent#5807]
## ✨ Feature

As a part of #5430, dark mode has been temporarily disabled so that automated themes could be enabled. Because styling would be automatically applied, even without a toggle, users with their browser preference set to dark mode would have triggered the styles, resulting in potentially inaccessible content.

A separate PR addressing dark mode styles (and thus re-enabling it) will be submitted to address this issue.

## 🙋‍♀️ Proposed Implementation

- Re-enable dark mode
- Review (and update) dark mode styles for components
- Review and optimize the `ThemeProvider`
- Review (and update) the theme switcher component so that it's ready for production
- Test all three mode options
- Apply the theme switcher component to the common header element

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] Every story should successfully be able to swap to dark mode
- [ ] Every story's dark mode implementation should be legible and accessible
- [ ] When viewing the application, selecting a mode using the theme switcher should trigger that mode
- [ ] When leaving the page and returning, the selected mode should be remembered and applied automatically
- [ ] Any issues have new tickets created

## :stop_sign: Blockers

List any issues which must be completed before this one.

```[tasklist]
### Blockers
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/pull/5430
```

**Comment by esizer** [comment]
> [!NOTE]
> Very few issues arose directly related to dark mode. Think I'll just resolve them in this ticket. 

## Review notes

### Public Pages

#### Job Poster

Colour contrast issue.

![Screenshot 2024-02-15 125354](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/c88a4b02-19ca-4b8b-9841-90104f7556ac)

### Admin Pages

#### Screening and assessment

Colour contrast issue,

![Screenshot 2024-02-15 133928](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/8c80bd55-6e72-4dea-80d2-c4b45bf11f09)

#### User Edit

Colour contrast issue on chip in roles dialog

![Screenshot 2024-02-15 134757](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/f3bd6f8b-21f1-46ca-b3d3-1db2c4238cc7)

**Comment by RM-1978** [comment]
> [!NOTE]
> Very few usability issues arose directly related to dark mode. @esizer, as we said, can they be resolved in this ticket? 

## Review notes

### Public Pages

#### Home Page

- [ ] Low perceptible  color contrast ratio: 
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/101275838/1f467345-f40a-4a4c-9b74-796ac42c3588)

#### Find Talent 

- [ ] Low perceptible  color contrast ratio: 
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/101275838/7435168d-224d-4798-85d9-ea47ac527328)
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/101275838/e5b69cee-00a0-425a-bc1c-dff425d32e35)

#### Job Poster

- [ ] Focus order issue - Tab order skips key concept between Education requirement cards. 
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/101275838/045a8574-6217-4857-865b-439503ec74db)
- [ ] Low perceptible  color contrast ratio:
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/101275838/2386dc0a-ed57-43f6-8727-83001853164c)

#### Profile and applications

- [ ] Low perceptible  color contrast ratio:
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/101275838/22c5e9a6-cb9f-42b4-8145-ccf476a9bd83)

### Admin Pages

#### Processes

- [ ] Low perceptible color contrast ratio:
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/101275838/21137bb9-3666-4ae6-9638-1e30151f3b97)

#### My Teams

- [ ] Low perceptible color contrast ratio:
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/101275838/5c2730b0-b1d3-4966-92b0-5f271af2b272)

#### Talent requests 
- [ ] Low perceptible color contrast ratio:
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/101275838/33e39bd5-b080-4b66-a2c7-8e296251c74a)

#### Skills

- [ ] Low perceptible color contrast ratio:
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/101275838/16aafad3-7aa6-4e01-b6ac-c7dec6a6b77a)

#### Screening and assessment
- [ ] Low perceptible color contrast ratio:
